tests/test_server_compiler.py

```diff
@@ -77,6 +84,142 @@ def test_server_compiler_compile_edgeql_script(self):
             ''',
         )
 
+    def _test_compile_structured_config(
+        self,
+        values: dict[str, Any],
+        *,
+        source: str = "config file",
+        **expected: Any,
+    ) -> dict[str, config.SettingValue]:
+        result = self.compiler.compile_structured_config(
+            {"cfg::Config": values}, source=source
+        )
+        rv = dict(result["cfg::Config"])
+        for name, setting in rv.items():
+            self.assertEqual(setting.name, name)
+            self.assertEqual(setting.scope, config.ConfigScope.INSTANCE)
+            self.assertEqual(setting.source, source)
+        self.assertDictEqual({k: v.value for k, v in rv.items()}, expected)
+        return rv
```

Comment:
Could we get an end-to-end test, also?

One thing I am concerned about is how INSERTed objects specified via structured config will interact with objects inserted manually via `CONFIGURE INSTANCE INSERT ...`.

The *best* option is for the sets of objects to get merged. I spent about a day trying to make that work, and came to the conclusion that it would probably take me closer to a week, and would be pretty hairy. This is why I introduced `EDGEDB_MAGIC_SMTP_CONFIG`, so we wouldn't need to handle it.

If manually inserted objects just fully override, that is probably acceptable behavior (though it wouldn't work for our SMTP config case, so we would still need EDGEDB_MAGIC_SMTP_CONFIG), but I think it will take a little care to make that work properly also. (If you INSERT an object and then RESET it away, probably the structured config one should come back? But I think as implemented it won't quite work.)

Anyway, I don't think this really *blocks* us, but I think we might need to disable object config for now, and stick with EDGEDB_MAGIC_SMTP_CONFIG for the case we need. (We should probably use the structured parser for EDGEDB_MAGIC_SMTP_CONFIG though!)
I think getting objects to work well enough to cover that case for us will be a decent amount of extra work, and not a priority yet.

Reply:
Right, I also left 2 similar questions in [the previous PR](https://github.com/edgedb/edgedb/pull/8059#issuecomment-2515142722)

Reply:
> We should probably use the structured parser for EDGEDB_MAGIC_SMTP_CONFIG

Fixed!